app/src/org/commcare/tasks/LogSubmissionTask.java

```diff
@@ -104,14 +108,16 @@ public static String getSubmissionUrl(SharedPreferences appPreferences) {
 
     @Override
     protected LogSubmitOutcomes doInBackground(Void... params) {
-        if (HiddenPreferences.isLoggingEnabled()) {

```

Comment:
Question: I think the reason we used to skip this class entirely if logging was disabled is that otherwise we create a DeviceLogReportRecord and a File that are empty (since the db had no logs to serialize) which we would never clean out since we used to only remove those records on log submission.

Is that outcome now not possible due to other structural changes? Otherwise I think we'd still need to skip serialization if Logging is totally disabled.